go-controller/pkg/factory/handler.go

```diff
@@ -92,7 +114,12 @@ type informer struct {
 func (i *informer) forEachQueuedHandler(f func(h *Handler)) {
 	i.RLock()
 	defer i.RUnlock()
+	sortHandlers := make(handlerSlice, 0, len(i.handlers))
 	for _, handler := range i.handlers {
+		sortHandlers = append(sortHandlers, handler)
+	}
+	sort.Sort(sortHandlers)
```

Comment:
I'm worried doing this for every event is going to hurt performance. I think it would be better to store the order in the data structure. You could use a two-dimensional map like map[int]map[uint64]*Handler, where int is the priority. Should be easy to index into since we always know the priority. Then for processing you can just do:

```
for x := 0; x <= minPriority; x++ {
    	for _, handler := range i.handlers[x] {
		f(handler)
	}
}
```

Reply:
yeah true, I admit I had this exact same thought of sorting every time, but was like ok... the largest number of handlers are for pods/namspaces/nodes at 15, so might be insignificant, but I like the nested map idea, I went with that.

Reply:
@tssurya what about storing a sorted slice, and inserting at the sorted index? eg https://stackoverflow.com/questions/42746972/golang-insert-to-a-sorted-slice/67216833#67216833

Reply:
hmm you mean like a slice of maps - insert based on the handler.priority will be something like this:
````
{
    h1[id1]{id:1, prio:0},
    h2[id2]{id:2, prio:0}, 
    h3[id3]{id:3, prio:0}, 
    h4[id4]{id:4, prio:1}, 
    h5[id5]{id:5, prio:2}, 
}
````

now if I get another prio 0 id6 handler we insert it at position 3 move positions 4 & 5 down...that could also work, for now I went with what Tim said...I can change this if this is more preferable..